packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -623,6 +623,8 @@ abstract class ResidentRunner {
   }
 
   Future<Null> processTerminalInput(String command) async {
+    // When terminal doesn't support line mode, '\n' can sneak into the input.
+    command = command.replaceAll('\n', '');
```

Comment:
Ah, of course, thanks! Somehow when I was looking for something like trim, I missed it. Not sure why...